pkg/apis/psmdb/v1/perconaservermongodbrestore_types.go

```diff
@@ -64,13 +68,68 @@ type PerconaServerMongoDBRestoreList struct {
 	Items           []PerconaServerMongoDBRestore `json:"items"`
 }
 
-func (r *PerconaServerMongoDBRestore) CheckFields() error {
+func (r *PerconaServerMongoDBRestore) CheckFields(cluster *PerconaServerMongoDB) error {
 	if len(r.Spec.ClusterName) == 0 {
 		return fmt.Errorf("spec clusterName field is empty")
 	}
-	if len(r.Spec.BackupName) == 0 && (len(r.Spec.StorageName) == 0 || len(r.Spec.Destination) == 0) {
-		return fmt.Errorf("fields backupName or storageName and destination is empty")
+
+	if r.Spec.PITR == nil {
+		if len(r.Spec.BackupName) == 0 && (len(r.Spec.StorageName) == 0 || len(r.Spec.Destination) == 0) {
+			return errors.New("fields backupName or storageName and destination is empty")
+		}
+
+		return nil
+	}
+
+	if !cluster.Spec.Backup.IsEnabledPITR() && r.Spec.BackupName == "" && r.Spec.StorageName == "" {
+		return errors.New("backup/storage name must be specified for point in time restore with pitr disabled in cluster")
+	}
+
+	switch r.Spec.PITR.Type {
+	case PITRestoreTypeDate:
+		if r.Spec.PITR.Date == nil {
+			return errors.New("date is required for pitr restore by date")
+		}
+
+	default:
+		return errors.Errorf("undefined pitr restore type: %s", r.Spec.PITR.Type)
+	}
+
+	return nil
+}
+
+type PITRestoreSpec struct {
+	Type PITRestoreType  `json:"type,omitempty"`
+	Date *PITRestoreDate `json:"date,omitempty"`
+}
+
+type PITRestoreType string
+
+var (
+	PITRestoreTypeDate PITRestoreType = "date"
+)
+
+type PITRestoreDate struct {
+	metav1.Time
+}
+
+func (t *PITRestoreDate) UnmarshalJSON(b []byte) (err error) {
+	if len(b) == 4 && string(b) == "null" {
```

Comment:
Probably we can use just `string(b) == "null"` check

Reply:
yes, this is optimization for large arrays, so we don't copy it (on a cast to string) without a need

Reply:
Make sense